app/controllers/api/backend_api_configs_controller.rb

```diff
@@ -34,7 +63,11 @@ def authorize
 
   delegate :backend_api_configs, to: :service
 
-  def backend_api_params
+  def backend_api_config_params
     params.require(:backend_api_config).permit(:backend_api_id, :path)
```

Comment:
I find this very dangerous... This accepts any existing backend_api_id even if that backend_api does not belong to the same tenant.
This is why I did and said this: https://github.com/3scale/porta/pull/1231#issuecomment-534463227
A safer approach for example could be:
```ruby
def backend_api_params
  backend_api_params = params.require(:backend_api_config).permit(:backend_api_id, :path)
  if (backend_api_id = backend_api_params.delete(:backend_api_id))
    backend_api_params[:backend_api] = current_account.backend_apis.find(backend_api_id)
  end
  backend_api_params
end
```
And of course no matter how you solve this issue, the fact of solving it requires a test because imo it is a big issue and easy to forget :smile:

Reply:
Btw, it is also better to return a 404 when it does not exist at all than a DB error raised for the foreign key being invalid :smile:

Reply:
Yes loading the object for 2 cases:

- it exists
- it belongs to the same tenant